.github/workflows/deploy_tre.yml

```diff
@@ -125,167 +123,110 @@ jobs:
-          # *************************************************************
-          export STATUS_CODE=$(curl -X "GET" "${TRE_URL}/api/workspace-templates/tre-workspace-innereye-deeplearning" -H "accept: application/json" -H "Authorization: Bearer ${TOKEN}" -k -s -w "%{http_code}" -o /dev/null)
-
-          if [[ ${STATUS_CODE} == 404 ]]
-          then
-            make register-bundle DIR=./templates/workspaces/innereye_deeplearning
-          fi
-
-          # Check if innereye_deeplearning_inference template is already registered
-          # ***********************************************************************
-          export STATUS_CODE=$(curl -X "GET" "${TRE_URL}/api/workspace-templates/tre-workspace-innereye-deeplearning-inference" -H "accept: application/json" -H "Authorization: Bearer ${TOKEN}" -k -s -w "%{http_code}" -o /dev/null)
-
-          if [[ ${STATUS_CODE} == 404 ]]
-          then
-            make register-bundle DIR=./templates/workspaces/innereye_deeplearning_inference
-          fi
-
-  publish_and_register_workspace_service_bundles:
-    name: Publish and Register Workspace Service Bundles
+  publish_and_register_bundles:
+    name: Register Bundle
     runs-on: ubuntu-latest
-    needs: [deploy_tre]
+    needs: [deploy_tre, prepare_bundle_registration]
+    strategy:
+      matrix:
+        include:
+        # bundles type can be inferred from the bundle dir (but this is more explicit)
+          - { BUNDLE_TYPE: "workspace", BUNDLE_DIR: "./templates/workspaces/base"}
+          - { BUNDLE_TYPE: "workspace", BUNDLE_DIR: "./templates/workspaces/azureml_devtestlabs"}
+          - { BUNDLE_TYPE: "workspace", BUNDLE_DIR: "./templates/workspaces/innereye_deeplearning"}
+          - { BUNDLE_TYPE: "workspace", BUNDLE_DIR: "./templates/workspaces/innereye_deeplearning_inference"}
+          - { BUNDLE_TYPE: "workspace_service", BUNDLE_DIR: "./templates/workspace_services/guacamole"}
+          - { BUNDLE_TYPE: "workspace_service", BUNDLE_DIR: "./templates/workspace_services/azureml"}
+          - { BUNDLE_TYPE: "workspace_service", BUNDLE_DIR: "./templates/workspace_services/devtestlabs"}
+          - { BUNDLE_TYPE: "workspace_service", BUNDLE_DIR: "./templates/workspace_services/innereye_deeplearning"}
+          - { BUNDLE_TYPE: "workspace_service", BUNDLE_DIR: "./templates/workspace_services/innereye_inference"}
     environment: Dev
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+
       - name: Azure Login
         uses: azure/login@v1
         with:
           creds: ${{ secrets.AZURE_CREDENTIALS }}
-      - name: Publish and register workspace bundles
+
+      - name: Install yq
+        run: |
+          sudo wget -O /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/v4.13.0/yq_linux_amd64
+          sudo chmod +x /usr/local/bin/yq
+
+      - name: Install porter
+        run: |
+          curl -L https://cdn.porter.sh/latest/install-linux.sh | bash
```

Comment:
```suggestion
          curl -L https://cdn.porter.sh/v0.38.6/install-linux.sh | bash
```

Reply:
I think we should use the same Porter version as is used by Resource Processor.

Reply:
Agree, will change this